src/main/java/cubicchunks/world/dependency/DependencyManager.java

```diff
@@ -0,0 +1,187 @@
+
+					// If the cube is supposed to reach a later stage, resume its generation.
+					if (requiredCube.getCube().getCurrentStage().precedes(requirement.getTargetStage())) {
+						this.cubeProvider.loadCube(requirement.getCoords(), LoadType.LOAD_OR_GENERATE, requirement.getTargetStage());
+					}
+				}
+			}
+			// Otherwise load it.
+			else {
+				this.cubeProvider.loadCube(requirement.getCoords(), LoadType.LOAD_OR_GENERATE, requirement.getTargetStage());
+			}
+		}
+	}
+
+	/**
+	 * Unregisters a given Dependent. Formerly required cubes will no longer be protected from being unloaded unless
+	 * other dependencies keep them around. The DependentCube will no longer be notified if one of its former
+	 * requirements advances a stage.
+	 *
+	 * @param dependent The DependentCube to be unregistered.
+	 */
+	public void unregister(Dependent dependent) {
+
+		// Remove all of the requirements.
+		Collection<Requirement> requirements = dependent.getRequirements();
+		for (Requirement requirement : requirements) {
+
+			// Get the required cube.
+			RequiredCube requiredCube = this.requiredMap.get(requirement.getCoords());
+
+			// Remove the dependent from the required cube's list of dependents.
+			requiredCube.removeDependent(dependent);
+
+			// If this was the last cube depending on the current required cube, forget about the required cube.
+			if (!requiredCube.isRequired()) {
+				this.requiredMap.remove(requirement.getCoords());
+			}
+		}
+	}
+
+	/**
+	 * Must be invoked whenever a cube, which has reached its target stage, has been loaded. Notifies all dependents of
+	 * the cube.
+	 *
+	 * @param cube The newly loaded cube.
+	 */
+	public void onLoad(Cube cube) {
+		RequiredCube requiredCube = this.requiredMap.get(cube.getCoords());
+		if (requiredCube != null) {
+			requiredCube.setCube(cube);
+			requiredCube.update(this);
+		}
+	}
+
+	/**
+	 * Checks if the cube at a given set of coordinates is currently required to be loaded.
+	 *
+	 * @return True iff there exists at least one dependent requiring the given cube to remain loaded.
+	 */
+	public boolean isRequired(CubeCoords coords) {
+		RequiredCube requiredCube = this.requiredMap.get(coords);
+		return requiredCube != null && requiredCube.isRequired();
+	}
+
+	/**
+	 * Must be invoked whenever a cube advances to a new stage. Notifies all dependents of the cube.
+	 *
+	 * @param cube The cube for which its dependents will be notified.
+	 */
+	public void updateDependents(Cube cube) {
+		RequiredCube requiredCube = this.requiredMap.get(cube.getCoords());
+		if (requiredCube != null) {
+			requiredCube.update(this);
+		}
+	}
+
+	/**
+	 * {@link #isRequired(CubeCoords coords)}
+	 *
+	 * @return True iff there exists at least one dependent requiring the given cube to remain loaded.
+	 */
+	public boolean isRequired(Cube cube) {
+		return this.isRequired(cube.getCoords());
+	}
+
+	public void tick() {
+		CubicChunks.LOGGER.info("{}", this.requiredMap.size());
```

Comment:
Does it really run every tick? If so, I would suggest changing log level to DEBUG.